pallets/manta-pay/src/lib.rs

```diff
@@ -480,19 +465,140 @@ pub mod pallet {
 	where
 		T: Config,
 	{
-		/// Returns the ledger implementation for this pallet.
+		/// Maximum Number of Updates per Shard
+		const PULL_MAX_PER_SHARD_UPDATE_SIZE: usize = 128;
+
+		/// Maximum Size of Sender Data Update
+		const PULL_MAX_SENDER_UPDATE_SIZE: usize = 1024;
+
+		/// Pulls receiver data from the ledger starting at the `receiver_index`.
 		#[inline]
-		fn ledger() -> Ledger<T> {
-			Ledger(PhantomData)
+		fn pull_receivers(receiver_index: &mut [usize; 256]) -> (bool, ReceiverChunk) {
+			let mut more_receivers = false;
+			let mut receivers = Vec::new();
```

Comment:
Can we create it like this `Vec::with_capacity(Self::PULL_MAX_PER_SHARD_UPDATE_SIZE)`?
Then `receivers.shrink_to_fit()' if it doesn't reach the reserved length.

Reply:
This is not a good idea for the typical use case, when the actual receiver size is small.